### PR TITLE
Added function to get a user's avatar URL

### PR DESCRIPTION
**Pull request by rxl881**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -397,6 +397,24 @@ func (cli *Client) SetDisplayName(displayName string) (err error) {
...
+func (cli *Client) GetAvatarURL() (url string, err error) {
+	urlPath := cli.BuildURL("profile", cli.UserID, "avatar_url")
+	s := struct {
+		AvatarURL string `json:"avatar_url"`
+	}{}
+	res, err := cli.MakeRequest("GET", urlPath, &s, nil)
+	if err != nil {
+		return "", err
+	}
+
```
Comment (Kegsay) [inline client.go]: Missing `defer res.Body.Close()`.

File: client_test.go [hunk 2]
```diff
@@ -24,6 +24,25 @@ func TestClient_LeaveRoom(t *testing.T) {
...
+		}
+		return nil, fmt.Errorf("unhandled URL: %s", req.URL.Path)
+	})
+
+	if response, err := cli.GetAvatarURL(); err != nil {
+		t.Fatalf("GetAvatarURL: Got error: %s", err.Error())
+	} else if response == "" {
+		t.Fatal("GetAvatarURL: Got empty response")
+	}
+
```
Comment (Kegsay) [inline client_test.go]: You should check what the `response` actually is (the MXC URI).

File: client.go [hunk 3]
```diff
@@ -397,6 +397,39 @@ func (cli *Client) SetDisplayName(displayName string) (err error) {
...
+		AvatarURL string `json:"avatar_url"`
+	}{url}
+	res, err := cli.MakeRequest("PUT", urlPath, &s, nil)
+	if err != nil {
+		return err
+	}
+
+	if err = json.Unmarshal(res, &s); err != nil {
+		return err
+	}
```
Comment (Kegsay) [inline client.go]: Why are you doing this? This is the PUT which doesn't return a JSON object with `avatar_url`..

**Comment by Kegsay** [comment]
Otherwise LGTM

File: client.go (hunk 3 above)
Comment (rxl881) [inline client.go]: Cheers for spotting this... I'm not quite sure where this came from ... possibly a leftover from  a copy and paste of GetAvatarURL, before I realised that MakeRequest could automatically marshal the response in to a passed object reference. Anyway, it should be fixed now.

**Comment by Kegsay** [comment]
LGTM!

**Pull request merged**